qa/sc_multiple_cert_submitters.py

```diff
@@ -0,0 +1,117 @@
+                                         sc_node_1_configuration, sc_node_2_configuration)
+        self.sc_nodes_bootstrap_info = bootstrap_sidechain_nodes(self.options, network)
+
+    def sc_setup_nodes(self):
+        return start_sc_nodes(self.number_of_sidechain_nodes, self.options.tmpdir)
+
+    def run_test(self):
+        time.sleep(0.1)
+
+        mc_node = self.nodes[0]
+        sc_node1 = self.sc_nodes[0]
+        sc_node2 = self.sc_nodes[1]
+
+        assert_true(sc_node1.submitter_isCertificateSubmitterEnabled()["result"]["enabled"], "sc node1 should be submitter")
+        assert_true(sc_node2.submitter_isCertificateSubmitterEnabled()["result"]["enabled"], "sc node2 should be submitter")
+
+        connect_sc_nodes(sc_node1, 1)  # Connect SC nodes
+
+        mcblock_hash1 = mc_node.generate(1)[0]
+        scblock_id1 = generate_next_blocks(sc_node1, "first node", 1)[0]
+        check_mcreferencedata_presence(mcblock_hash1, scblock_id1, sc_node1)
+
+        # Generate 8 more MC block to finish the first withdrawal epoch, then generate 1 more SC block to sync with MC.
+        we0_end_mcblock_hash = mc_node.generate(8)[7]
+
+        scblock_id2 = generate_next_block(sc_node1, "first node")
+        check_mcreferencedata_presence(we0_end_mcblock_hash, scblock_id2, sc_node1)
+
+        # Generate first mc block of the next epoch
+        we1_1_mcblock_hash = mc_node.generate(1)[0]
+        scblock_id3 = generate_next_blocks(sc_node1, "first node", 1)[0]
+        check_mcreference_presence(we1_1_mcblock_hash, scblock_id3, sc_node1)
+
+        # Wait until Certificate will appear in MC node mempool
+        time.sleep(10)
+        while (mc_node.getmempoolinfo()["size"] < 1 and (sc_node1.submitter_isCertGenerationActive()["result"]["state"]
+            or sc_node2.submitter_isCertGenerationActive()["result"]["state"])):
+            logging.info("Wait for certificate in mc mempool...")
+            print("node1 isGenerating: " + str(sc_node1.submitter_isCertGenerationActive()["result"]["state"]))
+            print("node2 isGenerating: " + str(sc_node2.submitter_isCertGenerationActive()["result"]["state"]))
+            time.sleep(2)
+        assert_equal(1, mc_node.getmempoolinfo()["size"], "Certificate was not added to Mc node mempool.")
+
+        # wait till both SC nodes stop generating certificate
+        while sc_node1.submitter_isCertGenerationActive()["result"]["state"] or sc_node2.submitter_isCertGenerationActive()["result"]["state"]:
+            time.sleep(2)
+
+        # Assert that proper log messages are present in both log files - only one node submitted certificate and the other one faced no errors
+        with open(os.path.join(self.options.tmpdir + "/sc_node0/log", "debugLog.txt")) as node0, open(os.path.join(self.options.tmpdir + "/sc_node1/log", "debugLog.txt")) as node1:
+            log0 = node0.read()
+            log1 = node1.read()
+            submission_error = "[ERROR]"
```

Comment:
Can we test a more specific string? "ERROR" can be also found for other reasons

Reply:
Instead of doing blocklisting of errors, I added a log message in CeasingSidechain when a certificate of equal or higher quality is already present in the current epoch, thus making the logging behavior of the two strategies as similar as possible.
I changed this python test to follow that approach, and now it checks if the two appropriate log messages are present, instead of checking if no error exists.